prom/poller.go

```diff
@@ -203,11 +220,28 @@ func suffixFor(m *dto.Metric) string {
 	result := make([]string, 0, len(m.Label))
 
 	for _, lp := range m.Label {
-		result = append(result, lp.GetName()+"_"+lp.GetValue())
+		labelName := strings.Map(charMapper, lp.GetName())
+		labelVal := strings.Map(charMapper, lp.GetValue())
+		result = append(result, labelName+"_"+labelVal)
 	}
 
 	if len(result) == 0 {
 		return ""
 	}
 	return "." + strings.Join(result, ".")
 }
+
+func charMapper(r rune) rune {
```

Comment:
does `||` have precedence over `&&` enough that this works the way we all think it does?